### PR TITLE
Update color picker according to UI Kit v5

### LINKED ISSUE #1360: Docs > Forms > Form Control > Color: Fix color picker default color
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

This issue comes from the work done in Issue https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1201 List in all the doc the incompatibility with ODS in which all Boosted components (variants included) have been inspected in order to identify the ones that do not respect ODS specifications and therefore need an alert message saying "Do not use in ODS".
Doing this work, design bugs were noticed.

**Design bug to be corrected:**

- docs/v5.1/forms/form-control/#color:  
    - [ ] Replace the default color of the color picker by an Orange branded color (color not decided yet)

### Reduced test cases

NA

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of Boosted are you using?

v5.1

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Description

Update color picker design.
Closes #1360 

### Motivation & Context

Compliance to UI Kit v5.

### Types of change

- New feature (non-breaking change which adds functionality)

### Live previews

- [https://deploy-preview-1794--boosted.netlify.app/docs/5.3/forms/form-control/#color](https://deploy-preview-1794--boosted.netlify.app/docs/5.3/forms/form-control/#color)

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- [x] My change follows accessibility good practices; I have at least run axe

#### Design

- [x] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- [x] My change is compatible with responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- (NA) I have added JavaScript unit tests to cover my changes

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] My change introduces changes to the migration guide
- (NA) My new component is added in Storybook
- (NA) My new component is compatible with RTL
- [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [x] Code review
- [x] Design review
- [x] A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by MewenLeHo** [comment]
Remaining edge case: we lose the box-shadow around the button when the panel of the color picker is open

File: scss/forms/_form-control.scss [hunk 1]
```diff
@@ -190,8 +190,23 @@ textarea {
 
 .form-control-color {
   width: $form-color-width;
-  height: $input-height;
-  padding: $input-padding-y;
+  height: $form-color-height; // Boosted mod
+  padding: $form-color-padding; // Boosted mod
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:194]: IMO, we could let the Bootstrap one here.

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
There's an issue once the color palette closes.
Missing disabled state.

File: scss/forms/_form-control.scss [hunk 2]
```diff
@@ -191,7 +191,24 @@ textarea {
...
+  border: 2px solid var(--#{$prefix}body-color);
+
+  // Boosted mod
+  &:focus {
+    &[data-focus-visible-added] {
+      background-color: $black;
+    }
+  }
+
+  &:not(:disabled):not([readonly]):hover {
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:204]: I'm not 100% sure but do we need to have such a complex selector since readonly doesn't seem to work on color input and less sure about this one but IMO, `:disabled` should have a `pointer-events: none` so doesn't need to be specified here (would need to add so on a disabled label `.is-disabled`).

File: scss/forms/_form-control.scss [hunk 3]
```diff
@@ -191,7 +191,24 @@ textarea {
...
+    background-color: var(--#{$prefix}body-color);
+  }
+
+  &:disabled {
+    filter: grayscale(1);
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:209]: Missing border-color change here (maybe need to be forced in `.form-control:disabled` but need to study the impact).
This filter could be changed to `filter: brightness(0) invert(1) brightness(0.4);`.

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
Few more changes

File: site/content/docs/5.2/forms/form-control.md [hunk 4]
```diff
@@ -115,8 +115,15 @@ If you want to have `<input readonly>` elements in your form styled as plain tex
...
 
 {{< example >}}
-<label for="exampleColorInput" class="form-label">Color picker</label>
-<input type="color" class="form-control form-control-color" id="exampleColorInput" value="#563d7c" title="Choose your color">
+<div class="mb-3">
+  <label for="exampleColorInput" class="form-label">Color picker</label>
+  <input type="color" class="form-control form-control-color" id="exampleColorInput" value="#a885d8" title="Choose your color">
+</div >
+
+<div class="mb-3">
```
[the remark this answers is not in the document]
Comment (MewenLeHo) [reply site/content/docs/5.2/forms/form-control.md:123]: Not sure to keep it either. We may keep it until we reach an agreement with the designers about the look of the disabled version.

**Review by Aniort — APPROVED** [review]
a11y ok

File: site/content/docs/5.2/migration.md [hunk 5]
```diff
@@ -219,6 +221,7 @@ Learn more by reading the new [color modes documentation]({{< docsref "/customiz
       <li><code>--bs-warning-bg-subtle</code></li>
       <li><code>--bs-warning-border-subtle</code></li>
       <li><code>--bs-warning-text</code></li>
+      <li><code>--bs-form-color-disabled-color</code></li>
```
Comment (louismaximepiton) [inline site/content/docs/5.2/migration.md:224]: Reorder alphabetically (same below)

File: scss/forms/_form-control.scss [hunk 6]
```diff
@@ -192,6 +192,26 @@ textarea {
   width: $form-color-width;
   height: $input-height;
   padding: $input-padding-y;
+  border-color: var(--#{$prefix}body-color);
+
+  // Boosted mod
+  &:not(:disabled):hover {
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:198]: Not sure that the `:not(:disabled)` is mandatory here.

File: scss/forms/_form-control.scss [hunk 7]
```diff
@@ -192,6 +192,26 @@ textarea {
   width: $form-color-width;
   height: $input-height;
   padding: $input-padding-y;
+  border-color: var(--#{$prefix}body-color);
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:195]: Missing Boosted mod

File: scss/forms/_form-control.scss [hunk 8]
```diff
@@ -31,7 +31,7 @@
   }
 
   // Customize the `:focus` state to imitate native WebKit styles.
-  &:focus {
+  &:not(.form-control-color):focus {
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:34]: Need a Boosted mod here too

File: scss/forms/_form-control.scss [hunk 9]
```diff
@@ -192,6 +192,26 @@ textarea {
...
+  &:disabled {
+    background-color: var(--#{$prefix}body-bg);
+    border-color: $form-color-disabled-color;
+
+    &::-moz-color-swatch {
+      filter: brightness(0) invert(1) brightness(.8);
+    }
+
+    &::-webkit-color-swatch {
+      filter: brightness(0) invert(1) brightness(.8);
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:211]: Should we introduce Sass variables everywhere for a more customizable color picker ?

File: scss/_variables.scss [hunk 10]
```diff
@@ -1091,7 +1091,8 @@ $input-line-height-lg:                  $h5-line-height !default; // Boosted mod
 
 $input-transition:                      border-color $transition-duration $transition-timing, $transition-focus !default;
 
-$form-color-width:                      3rem !default;
+$form-color-width:                      2.5rem !default; // Boosted mod
+$form-color-disabled-color:             $gray-500 !default; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_variables.scss:1095]:
```suggestion
$form-color-disabled-border-color:      $gray-500 !default; // Boosted mod

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
Few more things and one question before approving

File: scss/forms/_form-control.scss (hunk 6 above)
Comment (MewenLeHo) [reply scss/forms/_form-control.scss:198]: Not mandatory anymore since we found a way to change the background color for disabled state.

**Comment by Franco-Riccitelli** [comment]
I have made a design review of the colour picker and all is good. No comments.

File: site/content/docs/5.2/migration.md [hunk 11]
```diff
@@ -183,6 +185,8 @@ Learn more by reading the new [color modes documentation]({{< docsref "/customiz
       <li><code>--bs-emphasis-color</code></li>
       <li><code>--bs-emphasis-color</code></li>
       <li><code>--bs-form-check-bg</code></li>
+      <li><code>--bs-form-color-disabled-background-swatch</code></li>
+      <li><code>--bs-form-color-disabled-border-color</code></li>
```
Comment (louismaximepiton) [inline site/content/docs/5.2/migration.md:189]: Is there new CSS variables ?

**Review by louismaximepiton — COMMENTED** [review]
Almost feel right, forgot about it

**Review by Aniort — COMMENTED** [review]
In fact, vocalization on NVDA under Chrome is bad "cliquable  65% red 52% green 84% blue", so I need to investigate much more...

File: site/content/docs/5.2/forms/form-control.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (MewenLeHo) [reply site/content/docs/5.2/forms/form-control.md:123]: According to last meeting, we decided to keep the disabled version inside the documentation.

**Comment by MewenLeHo** [comment]
Please note that this pull request is linked to [Remove color picker borders on Chromium based browsers #37936](https://github.com/twbs/bootstrap/pull/37936).

**Comment by MewenLeHo** [comment]
It is also linked to the following issue [docs/5.2/forms/form-control/#color element is not supported in most UA and AT #37991](https://github.com/twbs/bootstrap/issues/37991)

File: site/content/docs/5.2/forms/form-control.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply site/content/docs/5.2/forms/form-control.md:123]: We might remove the `.mb-3` here

**Review by louismaximepiton — APPROVED** [review]
LGTM

File: site/content/docs/5.2/forms/form-control.md [hunk 12]
```diff
@@ -114,9 +114,20 @@ If you want to have `<input readonly>` elements in your form styled as plain tex
 
 Set the `type="color"` and add `.form-control-color` to the `<input>`. We use the modifier class to set fixed `height`s and override some inconsistencies between browsers.
 
+{{< callout info >}}
+{{< partial "callouts/warning-color-picker.md" >}}
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/form-control.md:118]: IMO I wouldn't use a partial if only used at one location. Moreover, the level of heading could change if it became reusable at several locations.

**Comment by Aniort** [comment]
a11y OK

**Review by Aniort — APPROVED** [review]
A11y warning text OK

File: scss/_variables.scss [hunk 13]
```diff
@@ -1091,7 +1091,9 @@ $input-line-height-lg:                  $h5-line-height !default; // Boosted mod
 
 $input-transition:                      border-color $transition-duration $transition-timing, $transition-focus !default;
 
-$form-color-width:                      3rem !default;
+$form-color-width:                      2.5rem !default; // Boosted mod
```
Comment (julien-deramond) [inline scss/_variables.scss:1094]:
```suggestion
$form-color-width:                      2.5rem !default; // Boosted mod: instead of `3rem`
```

Experimenting with the latest commits from Bootstrap this way of indicating whether a value has been changed from Bootstrap or is specific to Boosted.

File: scss/forms/_form-control.scss [hunk 14]
```diff
@@ -31,7 +31,7 @@
   }
 
   // Customize the `:focus` state to imitate native WebKit styles.
-  &:focus {
+  &:not(.form-control-color):focus { // Boosted mod
```
Comment (julien-deramond) [inline scss/forms/_form-control.scss:34]:
```suggestion
  &:not(.form-control-color):focus { // Boosted mod: instead of `&:focus`
```

File: site/content/docs/5.3/forms/form-control.md [hunk 15]
```diff
@@ -114,9 +114,20 @@ If you want to have `<input readonly>` elements in your form styled as plain tex
 
 Set the `type="color"` and add `.form-control-color` to the `<input>`. We use the modifier class to set fixed `height`s and override some inconsistencies between browsers.
 
+{{< callout info >}}
```
Comment (julien-deramond) [inline site/content/docs/5.3/forms/form-control.md:117]:
```suggestion
{{< callout warning >}}
```

File: site/content/docs/5.3/forms/form-control.md [hunk 16]
```diff
@@ -114,9 +114,20 @@ If you want to have `<input readonly>` elements in your form styled as plain tex
...
+**Accessibility warning** The support among browsers/AT is not robust enough (https://a11ysupport.io/tests/tech__html__input__input-color#assertion-html-input(type-color)_element-convey_name). So, use with care and provide an alternative way of entering or choosing a color value (eg. an input text fields to enter a hexadecimal color value).
```
Comment (julien-deramond) [inline site/content/docs/5.3/forms/form-control.md:118]:
```suggestion
Support among browsers/AT is not robust enough (https://a11ysupport.io/tests/tech__html__input__input-color#assertion-html-input(type-color)_element-convey_name). So, use with care and provide an alternative way of entering or choosing a color value (e.g., an input text field to enter a hexadecimal color value).
```

Suggestion to remove this title. We don't precise it usually. For example in `modal.md`:

```md
{{< callout warning >}}
While both ways to dismiss a modal are supported, keep in mind that dismissing from outside a modal does not match the [ARIA Authoring Practices Guide dialog (modal) pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/). Do this at your own risk.
{{< /callout >}}
```

Plus fixing typo.

File: site/content/docs/5.3/migration.md [hunk 17]
```diff
@@ -109,6 +109,8 @@ Learn more by reading the new [color modes documentation]({{< docsref "/customiz
 
 - <span class="badge bg-warning">Warning</span> Form text examples have been modified to add some precisions; form text should be explicitly associated with the form control it relates to using the `aria-labelledby` (for mandatory information such as data format) or `aria-describedby` (for complementary information) attribute. Please apply this modification in your websites if needed.
 
+- `.form-control-color` component has been redesigned according to UI Kit v5.
```
Comment (julien-deramond) [inline site/content/docs/5.3/migration.md:112]: I would drop it since it has no impact on users when migrating their website.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Looks good to me. Support is not that bad actually, works well from Safari v12 for example.
Few details to fix/discuss and we're good to go 🚀

File: site/content/docs/5.3/forms/form-control.md (hunk 16 above)
Comment (julien-deramond) [reply site/content/docs/5.3/forms/form-control.md]: @MewenLeHo What about removing the title removal + removal "The" at the beginning of the sentence?

**Comment by julien-deramond** [comment]
@MewenLeHo Please fix the bundlewatch issue as well

**Review by julien-deramond — APPROVED** [review]
